### DISCUSSION TITLE
Inline chat gives worse responses than given by regular copilot completions

### DISCUSSION THREAD

**Discussion by pokey** [community/community#66365] [Copilot Conversations]
### Select Topic Area

Product Feedback

### Body

I'd love to use the inline chat functionality to be able to ask Copilot to transform specific pieces of code. However, the responses seem to be far worse quality than the responses given by the regular Copilot code auto-completion suggestions. For example, here's inline chat:

<img width="1196" alt="image" src="https://github.com/community/community/assets/755842/35538c79-ea6e-4c56-b8f2-c1a6efebe333">

and here's auto-completion suggestions:

<img width="626" alt="image" src="https://github.com/community/community/assets/755842/0589dd1a-7b34-4ee5-9694-e50ec8548d86">

Note how the auto-completion suggestion leverages its knowledge of the surrounding codebase to know that `noSpokenForm` is a function, and how to call it.

From what I can tell, the auto-completion suggestions are based on sophisticated prompt engineering that gives the LLM context about your codebase, whereas the inline chat responses only seem to look at the selected text itself.

It would be great if the inline chat had similar prompt engineering as the autocompletion suggestions so that it is tailored to the surrounding context, as well as the broader codebase (eg past files you've viewed)

We would like to use the inline chat functionality to enable us to apply LLM-based transformations to text using [Cursorless](https://www.cursorless.org/) targets

**Comment by pkozlows** [comment]
I had a stroke and so I have a fine mortar impairment, so I am using the Cursorless. I am a physics undergraduate student and so I frequently use the Latex Workshop in vscode, for which Cursorless gives great support . I am using copilot to do latex, using the auto complete function; it would be great for the inline chat to work just as well, so I could use that to, but unfortunately it doesn’t. And example I had today:
```tex
\begin{equation}
\expval{\hat{x}^2} = \int_{0}^{\infty} \int_{0}^{\pi} \int_{0}^{2\pi} \psi_{n_x,n_y,n_z}^_(r,\theta,\phi) \hat{x}^2 \psi_{n_x,n_y,n_z}(r,\theta,\phi) r^2 \sin\theta \dd{r} \dd{\theta} \dd{\phi}
\end{equation}
Let’s simplify the integral:
```
autocomplete (good)>>
```tex
\begin{equation}
\expval{\hat{x}^2} = \frac{8}{L_x L_y L_z} \int_{0}^{\infty} \int_{0}^{\pi} \int_{0}^{2\pi} \sin\left(\frac{n_x \pi x}{L_x}\right) \sin\left(\frac{n_y \pi y}{L_y}\right) \sin\left(\frac{n_z \pi z}{L_z}\right) \hat{x}^2 \sin\left(\frac{n_x \pi x}{L_x}\right) \sin\left(\frac{n_y \pi y}{L_y}\right) \sin\left(\frac{n_z \pi z}{L_z}\right) r^2 \sin\theta \dd{r} \dd{\theta} \dd{\phi}
\end{equation}
```
However, I tried the same thing with the inline chat, and the result was worse:
```tex
\begin{equation}
\expval{\hat{x}^2} = \int_{0}^{\infty} \int_{0}^{\pi} \int_{0}^{2\pi} \psi_{n_x,n_y,n_z}^_(r,\theta,\phi) \hat{x}^2 \psi_{n_x,n_y,n_z}(r,\theta,\phi) r^2 \sin\theta \dd{r} \dd{\theta} \dd{\phi}
\end{equation}
```
>> 
<img width="522" alt="Screenshot 2023-10-24 at 8 51 09 AM" src="https://github.com/community/community/assets/59596755/ae88e51c-d2c8-4f5b-a19d-8e1da46084ab">
select prev line, cmd-i, ‘simplify this integral’
error in sc
Is it possible to make the inline chat experience better?
